panther_analysis_tool/schemas.py

```diff
@@ -154,44 +154,95 @@ def validate(
     ignore_extra_keys=False,
 )  # Prevent user typos on optional fields
 
-RULE_SCHEMA = Schema(
-    {
-        "AnalysisType": Or("rule", "scheduled_rule"),
-        "Enabled": bool,
-        Or("Filename", "Detection", only_one=True): Or(str, object),
-        "RuleID": And(str, NAME_ID_VALIDATION_REGEX),
-        Or("LogTypes", "ScheduledQueries", only_one=True): And([str], [LOG_TYPE_REGEX]),
-        "Severity": Or("Info", "Low", "Medium", "High", "Critical"),
-        Optional("Description"): str,
-        Optional("DedupPeriodMinutes"): int,
-        Optional("InlineFilters"): object,
-        Optional("DisplayName"): And(str, NAME_ID_VALIDATION_REGEX),
-        Optional("OnlyUseBaseRiskScore"): bool,
-        Optional("OutputIds"): [str],
-        Optional("Reference"): str,
-        Optional("Runbook"): str,
-        Optional("SummaryAttributes"): [str],
-        Optional("Threshold"): int,
-        Optional("Tags"): [str],
-        Optional("Reports"): {str: list},
-        Optional("Tests"): [
-            {
-                "Name": str,
-                Optional(
-                    "LogType"
-                ): str,  # Not needed anymore, optional for backwards compatibility
-                "ExpectedResult": bool,
-                "Log": object,
-                Optional("Mocks"): [MOCK_SCHEMA],
-            }
-        ],
-        Optional("DynamicSeverities"): object,
-        Optional("AlertTitle"): str,
-        Optional("AlertContext"): object,
-        Optional("GroupBy"): object,
-    },
-    ignore_extra_keys=False,
-)  # Prevent user typos on optional fields
+
+def rule_validator(d):
+    if "BaseDetection" in d:
+        derived_forbidden_fields = {"Tests", "LogTypes", "Detection"}
+        for field in derived_forbidden_fields:
+            if field in d:
+                raise SchemaError(f"'{field}' is forbidden when 'BaseDetection' is present.")
+    else:
+        base_required_fields = {
+            "Enabled",
+            "Severity",
+            "Filename",
+            "Detection",
+            "LogTypes",
+            "ScheduledQueries",
+        }
+        for field in base_required_fields:
+            if field not in d:
+                raise SchemaError(f"'{field}' is required when 'BaseDetection' is not present.")
```

Comment:
I wouldn't even mention BaseDetection in this context.

Reply:
thanks. removing all this.